src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    let newState = Object.assign({}, currentState);
```

Comment:
Don't need to create a new state here, because you already have a copy on 12 link
```suggestion
```